app/com/yetu/youtube/controllers/YoutubeController.scala

```diff
@@ -53,7 +54,21 @@ class YoutubeController @Inject()(implicit val env: Environment[User, SessionAut
     for {
       info: Option[OAuth2Info] <- oauth2Dao.find(request.identity.loginInfo)
       accessToken: String = info.map(_.accessToken).getOrElse("Invalid access token")
-      wsResponse <- InboxService.sendToInbox(request.body, accessToken)
+      wsResponse <- InboxService.sendToInbox(request.body, accessToken, youtubeEventName)
+      response = wsResponseToPlayResponse(wsResponse)
+    } yield response
+
+  }
+
+	/**
+   * Handles the "general notification" request, this will be moved to a separate application one day
+   */
+  def notification = SecuredAction.async(parse.json) { implicit request =>
+
+    for {
+      info: Option[OAuth2Info] <- oauth2Dao.find(request.identity.loginInfo)
+      accessToken: String = info.map(_.accessToken).getOrElse("Invalid access token")
+      wsResponse <- InboxService.sendToInbox(request.body, accessToken, notificationEventName)
```

Comment:
the two actions share 95% of the code, the only difference is the event name. Could you put this code in a function to reduce code duplication?